docs/workforce-management-framework/topics/pro-keycloak-implementation.adoc

```diff
@@ -0,0 +1,54 @@
+[id='pro-keycloak-implementation-{chapter}']
+= Implementing {WFM-RC-KeycloakName}
+
+This section includes:
+
+. xref:pro-initializing-keycloak-{chapter}[Initializing {WFM-RC-KeycloakName}]
+. xref:pro-importing-the-endpointsecurity-interface-{chapter}[Importing the EndpointSecurity Interface]
+. xref:pro-importing-keycloak-init-{chapter}[Importing {WFM-RC-KeycloakName} _init_]
+. xref:pro-applying-keycloak-init-{chapter}[Applying {WFM-RC-KeycloakName} _init_ ]
+. xref:pro-enabling-keycloak-route-protection-{chapter}[Enabling {WFM-RC-KeycloakName} Route Protection]
+
+== Prerequisites
+
+. Ensure you have completed xref:getting-started[Getting Started]
+. Docker (tested with version: {WFM-RC-DockerVersion})
+
+== Using {WFM-RC-KeycloakName} to secure endpoints
+
+[id='pro-initializing-keycloak-{chapter}']
+[discrete]
+=== Initializing {WFM-RC-KeycloakName}
+
+See the link:{WFM-RC-CoreURL}{WFM-RC-Branch}/demo/server/src/modules/keycloak/index.ts[Keycloak code] for reference.
+
+[id='pro-importing-the-endpointsecurity-interface-{chapter}']
+[discrete]
+=== Importing The EndpointSecurity Interface
+
+Import the link:{WFM-RC-CoreURL}{WFM-RC-Branch}/cloud/auth/src/EndpointSecurity.ts[EndpointSecurity Interface] and apply it to a variable <ENG: what variable and how?>.
+
+[id='pro-importing-keycloak-init-{chapter}']
+[discrete]
+=== Importing {WFM-RC-KeycloakName} _init_
+
+Import *{WFM-RC-KeycloakName}* _init_ <ENG: how?> from link:{WFM-RC-CoreURL}{WFM-RC-Branch}/demo/server/src/modules/keycloak/index.ts[Keycloak].
+
+[id='pro-applying-keycloak-init-{chapter}']
+[discrete]
+=== Applying {WFM-RC-KeycloakName} _init_
+
+Apply *{WFM-RC-KeycloakName}* _init_ to the security Middleware <ENG: what is 'security Middleware' and how?> to enable *{WFM-RC-KeycloakName}* for route protection.
+See the link:{WFM-RC-CoreURL}{WFM-RC-Branch}/demo/server/src/modules/index.ts[securitySetup] code for more information.
+
```

Comment:
all info is provided in the linkes `securitySetup` file
middleware https://github.com/feedhenry-raincatcher/raincatcher-core/blob/983cd8c607cd417975acca86b1eb7d71dc40b039/demo/server/src/modules/index.ts#L35

how to enable : https://github.com/feedhenry-raincatcher/raincatcher-core/blob/983cd8c607cd417975acca86b1eb7d71dc40b039/demo/server/src/modules/index.ts#L55
These concepts should be clear for people with express.js basic experience (>99% of node devs)